### ISSUE TITLE
AesExtraDataRecord not present in local header for aes encrypted data

### ISSUE THREAD

**Issue by Windolian** [srikanth-lingala/zip4j#96]
I have this error when unzipping a zip file inside another zip file (both with AES-256 password encryption).
They work if I use zipcrypto instead of AES-256.
I zipped them with 7-zip.

```
java.lang.RuntimeException: AesExtraDataRecord not present in local header for aes encrypted data
	at net.lingala.zip4j.util.Zip4jUtil.getCompressionMethod(Zip4jUtil.java:92)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getCompressedSize(ZipInputStream.java:290)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.initializeEntryInputStream(ZipInputStream.java:199)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:106)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:79)
```

**Comment by srikanth-lingala** [comment]
I made a fix for it. Can you try with `2.2.4-SNAPSHOT`? 

Please also note that zip4j was not tested before with nested zip files. I would be glad to fix any issues that come up, but I just want to give out a disclaimer that this has not be tested so far.

But don't get me wrong: I would be glad if you can test this further and report any issues you find.

**Comment by Windolian** [comment]
> 
> 
> I made a fix for it. Can you try with `2.2.4-SNAPSHOT`?
> 
> Please also note that zip4j was not tested before with nested zip files. I would be glad to fix any issues that come up, but I just want to give out a disclaimer that this has not be tested so far.
> 
> But don't get me wrong: I would be glad if you can test this further and report any issues you find.

Thanks, you are awesome.

Now this error appears

```
 java.util.zip.DataFormatException: invalid distance too far back
java.io.IOException: java.util.zip.DataFormatException: invalid distance too far back
	at net.lingala.zip4j.io.inputstream.InflaterInputStream.read(InflaterInputStream.java:52)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.read(ZipInputStream.java:155)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.read(ZipInputStream.java:127)
```

**Comment by srikanth-lingala** [comment]
I am able to reproduce this with another zip file I generated. Will look into it.

**Comment by srikanth-lingala** [comment]
This is a tricky one. I have identified the issue, and a fix is on the way. Fix is also a bit complicated, so it might take a while. This fix especially needs a lot of testing.

**Comment by srikanth-lingala** [comment]
Luckily I managed to find a simpler fix for this issue. Can you please try with `2.2.4.1-SNAPSHOT`?

**Comment by Windolian** [comment]
Working perfectly. 
Thanks

**Comment by srikanth-lingala** [comment]
Fixed in v2.2.4 released today
